Number inputs now have + and - buttons instead of the default spinners

i-hardy opened the pull request:
#31 The buttons use the same styles as other input elements.

crawfoal commented on app/assets/javascripts/form_number_fields.coffee:
```diff
@@ -0,0 +1,30 @@
+$(document).on 'turbolinks:load', ->
+  $('input[type="number"]').each ->
+    input = $(this)
+    $('<div class="quantity-nav"><div class="quantity-button quantity-up">+</div><div class="quantity-button quantity-down">-</div></div>').insertAfter(input)
```
Instead of inserting extra HTML via JavaScript, we should create a helper method that can be called in the views. E.g. in `app/views/deeds/_form.html.erb`, we currently have

```
...
  <%= f.number_field :miles, min: 0, step: 1 %>
...
```

And we should create a helper method, say `custom_number_field`, that can be used like this:

```
...
  <%= f.custom_number_field :miles, min: 0, step: 1 %>
...
```

`f` is a `FormBuilder` object, and [the Rails guides discuss how to extend this object](http://guides.rubyonrails.org/form_helpers.html#customizing-form-builders).

Unlike their example, we want all of our forms to have this method. Actually, it would be even better if we could just wrap the standard Rails `number_field` method and add that HTML in. Setting this up is a little advanced, but I have a pretty good idea of how we can go about it. It's a little too involved to describe here, but if you'd like we can do a virtual paired programming session to get things set up. Or, if you feel confident making the changes feel free to get started 😃  A third option would be that I can implement the basics and then you can pick back up where I leave off.

crawfoal commented on app/assets/javascripts/form_number_fields.coffee:
```diff
@@ -0,0 +1,31 @@
+$(document).on 'turbolinks:load', ->
+  $('input[type="number"]').each ->
+    input = $(this)
+    $('<div class="quantity-nav">
```
Instead of inserting extra HTML via JavaScript, we should create a helper method that can be called in the views. E.g. in `app/views/deeds/_form.html.erb`, we currently have

```
...
  <%= f.number_field :miles, min: 0, step: 1 %>
...
```

And we should create a helper method, say `custom_number_field`, that can be used like this:

```
...
  <%= f.custom_number_field :miles, min: 0, step: 1 %>
...
```

`f` is a `FormBuilder` object, and [the Rails guides discuss how to extend this object](http://guides.rubyonrails.org/form_helpers.html#customizing-form-builders).

Unlike their example, we want all of our forms to have this method. Actually, it would be even better if we could just wrap the standard Rails `number_field` method and add that HTML in. Setting this up is a little advanced, but I have a pretty good idea of how we can go about it. It's a little too involved to describe here, but if you'd like we can do a virtual paired programming session to get things set up. Or, if you feel confident making the changes feel free to get started 😃  A third option would be that I can implement the basics and then you can pick back up where I leave off.

(updated to place this comment on the changes in the most recent commit)

i-hardy commented:
Hi, sorry for the slight delay, been busy with job interviews. I've managed to slim the coffeescript right down, there's now only a single event listener that utilises the '+' and '-' values of the buttons. I've also wrapped the Rails number_field method to return the custom buttons- this is currently in application_helper.rb as I couldn't seem to get it to work from a separate file, but I'm happy to move it out for tidiness.

i-hardy commented:
I've restyled the + and - buttons- they now sit either side of the number field, as they were too scrunched when on top of each other at the side, and they only have a border on one side so the field focus looks normal again! I've also cleaned up the coffeescript- eval definitely wasn't the best choice, so I've replaced it with a switch based on the operator- keeps the function small but avoids the issues with eval.

crawfoal commented:
Did you have a chance to check out that [example](https://github.com/crawfoal/greensteps/commit/4585cb3564282db6f93337388732264d95daf169) I mentioned? If it looks good to you, I'd like to incorporate it into this PR.

crawfoal commented:
Hi @i-hardy 🙂  I hope you don't mind, but I'm going to label this as "abandon" so that people know it's fair game to pick it up and complete it. If you'd like to start back up with it and finish it, just let me know and I'll remove the label.
